File: gemini/clients/node/test-connection.js
/**
 * Test Gemini API connection
 */

const { GeminiClient } = require('./gemini-client');

async function testConnection() {
  console.log('🧪 Testing Gemini API Connection...\n');

  try {
    const client = new GeminiClient();

    console.log('✅ Client initialized successfully');
    console.log(`📋 Default Model: ${client.defaultModel}\n`);

    // Test basic generation
    console.log('Testing basic text generation...');
    const response = await client.generateContent({
      prompt: 'Say "Hello from Gemini!" and nothing else.'
    });

    console.log('✅ Response received:');
    console.log(`📝 ${response.text}\n`);

    // Test token counting
    console.log('Testing token counting...');
    const tokens = await client.countTokens('This is a test message');
    console.log(`✅ Token count: ${tokens}\n`);

    // Display available models
    console.log('Available models:');
    const models = client.getAvailableModels();
    models.slice(0, 3).forEach(model => {
      console.log(`  • ${model.name} (${model.id})`);
    });

    console.log('\n🎉 All tests passed! Gemini integration is ready.');
    console.log('\nNext steps:');
    console.log('  1. Run examples: npm run example:chat');
    console.log('  2. Try streaming: npm run example:stream');
    console.log('  3. Test vision: npm run example:vision');

  } catch (error) {
    console.error('❌ Test failed:', error.message);
    console.error('\nTroubleshooting:');
    console.error('  1. Check that GEMINI_API_KEY is set in .env file');
    console.error('  2. Verify API key is valid at https://makersuite.google.com');
    console.error('  3. Ensure you have internet connection');
    process.exit(1);
  }
}

testConnection();

File: gemini/clients/python/test_connection.py
"""
Test Gemini API connection
"""

from gemini_client import GeminiClient


def test_connection():
    print('🧪 Testing Gemini API Connection...\n')

    try:
        client = GeminiClient()

        print('✅ Client initialized successfully')
        print(f'📋 Default Model: {client.default_model}\n')

        # Test basic generation
        print('Testing basic text generation...')
        response = client.generate_content('Say "Hello from Gemini!" and nothing else.')

        print('✅ Response received:')
        print(f'📝 {response["text"]}\n')

        # Test token counting
        print('Testing token counting...')
        tokens = client.count_tokens('This is a test message')
        print(f'✅ Token count: {tokens}\n')

        # Display available models
        print('Available models:')
        models = client.get_available_models()
        for model in models[:3]:
            print(f'  • {model["name"]} ({model["id"]})')

        print('\n🎉 All tests passed! Gemini integration is ready.')
        print('\nNext steps:')
        print('  1. Try the chat example')
        print('  2. Test streaming responses')
        print('  3. Experiment with vision analysis')

    except Exception as e:
        print(f'❌ Test failed: {e}')
        print('\nTroubleshooting:')
        print('  1. Check that GEMINI_API_KEY is set in .env file')
        print('  2. Verify API key is valid at https://makersuite.google.com')
        print('  3. Ensure you have internet connection')
        print('  4. Install dependencies: pip install google-generativeai python-dotenv')
        return 1

    return 0


if __name__ == '__main__':
    exit(test_connection())
